src/components/calcite-popover/calcite-popover.tsx

```diff
@@ -250,6 +242,26 @@ export class CalcitePopover {
   //
   // --------------------------------------------------------------------------
 
+  setUpReferenceElement = async (): Promise<void> => {
+    this.removeReferences();
+    this._referenceElement = this.getReferenceElement();
+
+    if (!this._referenceElement) {
+      await new Promise((resolve) => setTimeout(resolve, 250));
```

Comment:
Where does the 250 come from?

Reply:
It's just an arbitrary number. That's the thing. If the reference element isn't in the DOM, how long should we wait to try again?

Reply:
Maybe we should just remove this and have the warning so at least a user can see something happened?

Reply:
Yeah, that sounds like a good idea. There's no guarantee it'll be available after the timeout either. 😞  

I would've thought that resolving the reference element would be reliable by the time `open` is set to true. Does this problem show up only when the component is created w/ `open=true` and the ref element may not be there yet or is this problem occurring regardless of when the tooltip/popover is opened?

Reply:
Waiting for `open` isn't a since the `calcite-tooltip-manager` queries by `data-calcite-tooltip-reference` which doesn't get set unless a reference element is found. So it can't access the tooltip to tell it to find it's reference element because it identifies it by the data attribute.

Reply:
I see. Looks like we would need to refactor the manager + tooltip to cover this case. Is it recommended to have all tooltips after the tooltip manager? If so, maybe the warning could have some info about this. 

Another thing to try would be to resolve the reference element in the next animation frame to ensure the DOM is in sync. WDYT?

Reply:
> Another thing to try would be to resolve the reference element in the next animation frame to ensure the DOM is in sync. WDYT?

How would we do this?

Reply:
> Is it recommended to have all tooltips after the tooltip manager? If so, maybe the warning could have some info about this.

We could recommend that.

Reply:
> Looks like we would need to refactor the manager + tooltip to cover this case. 

We could refactor the entire tooltip/popover so that instead of using a referenceElement, the user slots the "trigger". However, that would mean that the tooltip surrounds the trigger and we lose the flexibility of having the tooltip and trigger be in completely different areas of the DOM.

Maybe we can just offer a slot as another way to set the reference element?

- Add `trigger` slot
- rename `reference-element` to `trigger-id`? Maybe a better name than `trigger`?

Reply:
@jcfranco Can we merge this so there's a warning at least? Then we can discuss the possible refactoring later?

Reply:
Sounds good. WRT the commit type/message, looks like there was more done than showing a warning when the ref is not found. If this is the case, can you revisit the PR title?

Reply:
Done